diff --git a/Cargo.toml b/Cargo.toml
index 07b1a66..9295680 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -56,7 +56,7 @@ halo2_proofs = { version = "0.3", optional = true, default-features = false, fea
 getset = "0.1"
 
 # Logging
-tracing = { version = "0.1", default-features = false }
+#tracing = { version = "0.1", default-features = false }
 
 # No-std support
 core2 = { version = "0.3", default-features = false, features = ["alloc"] }
